### PR TITLE
Reduce the number of compiler warnings when building.

### PR DESCRIPTION
**Pull request by Don-Ward**

### CODE REVIEW

File: src/iconc/ccomp.c [hunk 1]
```diff
@@ -173,11 +173,15 @@ char *srcname;
 char *exename;
    {
    struct lib *l;
-   char sbuf[MaxFileName];		/* file name construction buffer */
    int  rv, buflen;
-   char *buf, objname[MaxFileName];
+   char *buf;
    char *s;
-   char *dlrgint;
+   /* dlrgint disabled for now */
```
Comment (Jafaral) [inline src/iconc/ccomp.c:179]: I started to mark places like this with TODO or FIXME for easy grepping. Some IDEs also know how track those lables

File: src/iconc/ccomp.c [hunk 2]
```diff
@@ -173,11 +173,15 @@ char *srcname;
...
+#if 0
```
Comment (Jafaral) [inline src/iconc/ccomp.c:180]: I started to mark places like this with TODO or FIXME for easy grepping. Some IDEs also know how track those labels

File: src/iconc/inline.c [hunk 3]
```diff
@@ -850,6 +852,7 @@ char *typ;
    cd_add(cd);
    }
 
+#if 0
```
Comment (Jafaral) [inline src/iconc/inline.c:855]: TODO

File: src/iconc/tv.c [hunk 4]
```diff
@@ -840,6 +839,7 @@ tv_type_bits_set(tv, typcd)
       tv_bit_set(tv, i);
 }
 
+#if 0
```
Comment (Jafaral) [inline src/iconc/tv.c:842]: TODO

File: src/iconc/tv.c [hunk 5]
```diff
@@ -1082,6 +1083,7 @@ hash_bh(raw)
    return (u & hash_mask);
 }
 
+#if 0
```
Comment (Jafaral) [inline src/iconc/tv.c:1086]: TODO

File: src/runtime/fmisc.r [hunk 6]
```diff
@@ -2106,9 +2105,9 @@ function{*} keyword(keyname,ce,i)
    body {
       struct progstate *p = BlkD(d,Coexpr)->program;
       char *kname = kyname;
+#if 0
```
Comment (Jafaral) [inline src/runtime/fmisc.r:2108]: TODO

**Review by cjeffery — APPROVED** [review]
Looks OK-ish.

**Pull request merged**